js/src/forum/utils/PostControls.js

```diff
@@ -128,14 +129,22 @@ export default {
   },
 
   /**
-   * Hide a post.
-   *
-   * @return {Promise}
+   * Hide a post. (aka soft delete)
    */
   hideAction() {
-    this.pushAttributes({ hiddenAt: new Date(), hiddenUser: app.session.user });
-
-    return this.save({ isHidden: true }).then(() => m.redraw());
+    const confirmModal = new ConfirmModal();
+    confirmModal.init(
+      app.translator.trans('core.forum.post_controls.hide_post_modal.title'),
+      app.translator.trans('core.forum.post_controls.hide_post_modal.message'),
+      app.translator.trans('core.forum.post_controls.hide_post_modal.positive'),
+      app.translator.trans('core.forum.post_controls.hide_post_modal.negative'),
+      () => {
+        this.pushAttributes({ hiddenAt: new Date(), hiddenUser: app.session.user });
+  
+        return this.save({ isHidden: true }).then(() => m.redraw());
+      }
+    );
+    app.modal.show(confirmModal);
```

Comment:
This is not how we initialize modals in Flarum (at least in core). Apart from that, there's no reason to be passing translator messages.

```suggestion
    app.modal.show(
	    new ConfirmModal({
	    	action: 'hide',
	    	save: () => {
		    	this.pushAttributes({ hiddenAt: new Date(), hiddenUser: app.session.user });
		  
		        return this.save({ isHidden: true }).then(() => m.redraw());
	        }
	    })
    );
```

The translated messages should be translated directly in the children, not passed as properties.
To get this `save` callback, you can do `this.props.save()` (or another name if you rename it), and the action with `this.props.action` (for example).

Same in the `deleteAction` below.

This can then be used as 
```jsx
Button.component({
	className: 'Button Button--block',
	children: app.translator.trans(`core.forum.post_controls.${action}_post_modal.negative`),
    onclick: this.hide.bind(this),  
    disabled: this.loading // Disable negative button when loading
})
```

Reply:
> Apart from that, there's no reason to be passing translator messages.

Ok, but what if it will be used outside `post_controls`? Is it ok to pass translation path like that:
```js
new ConfirmModal({
  action: 'hide',
  translation: 'core.forum.post_controls.hide_post_modal',
  // ...
});
```

Reply:
Oh, I didn't realize the Modal was supposed to be kind of a global utility, I see. Uh, not sure what the other devs' thoughts are on this, but in that case I guess you would have to pass multiple translations. _Or_ you could have a function that returns the translation key like
```js
(action) => `core.forum.post.controls.{action}_post_modal`
```
but this is probably overcomplicating it.

Reply:
I didn't understand your idea with that function, because it'll be the same as the first message there. Anyway, I'll commit it like I said before, because I also realized it would be also great to have it in discussion hiding/deleting. It will look like this for now:
```js
hideAction() {
  app.modal.show(new ConfirmModal({
    translation: "core.forum.discussion_controls.hide_discussion_modal",
    save: () => {
      this.pushAttributes({ hiddenAt: new Date(), hiddenUser: app.session.user });
  
      return this.save({ isHidden: true });
    }
  }));
}
```